### PR TITLE
Adding disallowed_headers in the ext_proc header forwarding rule.

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
Adding disallowed_headers in the ext_proc header forwarding rule.
This is to specifically disallow certain headers to be forwarded to the ext_proc server.

This PR is a follow up PR of https://github.com/envoyproxy/envoy/pull/27333.  It addresses the API changes to add the disallowed_headers.  The actual implementation will follow later.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto [hunk 1]
```diff
@@ -204,9 +204,21 @@ message ExternalProcessor {
 
 // The HeaderForwardingRules structure specifies what headers are
 // allowed to be forwarded to the external processing server.
+// This works as below:
+// 1. If neither ``allowed_headers`` or ``disallowed_headers`` is set, all headers are forwarded.
```
Comment (tyxia) [inline api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto:208]: nit
```suggestion
// 1. If neither ``allowed_headers`` nor ``disallowed_headers`` is set
```

**Review by tyxia — APPROVED** [review]
LGTM, Thanks!

File: api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto (hunk 1 above)
Comment (yanjunxiang-google) [reply api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto]: done

**Review by mattklein123 — APPROVED** [review]
LGTM

**Pull request merged**